wfe/wfe.go

```diff
@@ -499,64 +502,80 @@ func (wfe *WebFrontEndImpl) verifyPOST(
 	ctx context.Context,
 	logEvent *requestEvent,
 	request *http.Request,
-	kx keyExtractor) ([]byte, *jose.JSONWebKey, *acme.ProblemDetails) {
+	kx keyExtractor) ([]byte, string, *jose.JSONWebKey, *acme.ProblemDetails) {
 
 	if prob := wfe.validPOST(request); prob != nil {
-		return nil, nil, prob
+		return nil, "", nil, prob
 	}
 
 	if request.Body == nil {
-		return nil, nil, acme.MalformedProblem("no body on POST")
+		return nil, "", nil, acme.MalformedProblem("no body on POST")
 	}
 
 	bodyBytes, err := ioutil.ReadAll(request.Body)
 	if err != nil {
-		return nil, nil, acme.InternalErrorProblem("unable to read request body")
+		return nil, "", nil, acme.InternalErrorProblem("unable to read request body")
 	}
 
 	body := string(bodyBytes)
 	parsedJWS, err := wfe.parseJWS(body)
 	if err != nil {
-		return nil, nil, acme.MalformedProblem(err.Error())
+		return nil, "", nil, acme.MalformedProblem(err.Error())
 	}
 
 	pubKey, prob := kx(request, parsedJWS)
 	if prob != nil {
-		return nil, nil, prob
+		return nil, "", nil, prob
 	}
 
 	return wfe.verifyJWS(pubKey, parsedJWS, request)
 }
 
-func (wfe *WebFrontEndImpl) verifyJWS(
+// Checks parsed JWS whether it matches the given public key, does some general
+// checks, and extracts the URL header parameter.
+func (wfe *WebFrontEndImpl) verifyJWSLight(
 	pubKey *jose.JSONWebKey,
 	parsedJWS *jose.JSONWebSignature,
-	request *http.Request) ([]byte, *jose.JSONWebKey, *acme.ProblemDetails) {
+	jwsPrefix string) ([]byte, string, *acme.ProblemDetails) {
```

Comment:
Are you sure that you need the `jwsPrefix` argument that you're adding to this function? It looks like `verifyJWS` always sets it to `""` and only `wfe.KeyRollover` uses it. I think it would be better to remove this argument and have `wfe.KeyRollover` make a new malformed problem with the correct message prefix when there is a non-nil err response from calling `verifyJWSLight` (or whatever it ends up being called!)